hudi-utilities/src/test/java/org/apache/hudi/utilities/testutils/UtilitiesTestBase.java

```diff
@@ -410,6 +411,16 @@ public static String toJsonString(HoodieRecord hr) {
       return records.stream().map(Helpers::toJsonString).toArray(String[]::new);
     }
 
+    public static Tuple2<String, String>[] jsonifyRecordsByPartitions(List<HoodieRecord> records, int partitions) {
+      Tuple2<String, String>[] data = new Tuple2[records.size()];
+      for (int i = 0; i < records.size(); i++) {
+        int key = i % partitions;
```

Comment:
shouldn't it be better with org.apache.hudi.common.model.HoodieKey#recordKey from each HoodieRecord ?